### ISSUE TITLE
How to get it to work with WiringPi?

### ISSUE THREAD

**Issue by gevzak** [mnestor/docker-swarm-devices#2]
Hey and thanks for the nice project.
I have been struggling to get device access to my docker container, which has to run on on a cluster of raspberry Pi's.
The issue is that sensors that are attached to them provide libraries, which need access to /dev and/or /dev/gpiomem.I have tried to get it work using your solution, but writingPi still refuses to work. My config files and logs are below.
I will appreciate if you can have a look and tell me if it's possible to get it working or not.
Thank you.

---------------Starting your container-------------------------------
WARNING: The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Starting app_volumes_1 ... done
Attaching to app_volumes_1
volumes_1  | Starting event listener container
volumes_1  | Listening for startup events that have a label of 'volume.device'
volumes_1  | Setting permissions (c 1:1 rwm) for f50417f072e405009a43bf4b35a1a4a6ca8013eee5b7ad94bb76e6cf5bdf7c00 to device (/dev/mem)

---------------Starting mycontainer which requires access to /dev/mem -------------------------------

`WARNING: The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Creating new_iot_1 ... done
Attaching to new_iot_1
iot_1  | wiringPiSetup: Unable to open /dev/mem or /dev/gpiomem: No such file or directory.
iot_1  |   Aborting your program because if it can not access the GPIO
iot_1  |   hardware then it most certianly won't work
iot_1  |   Try running with sudo?
new_iot_1 exited with code 1`

---------------my docker-compose.yml------------------------------
services:
  iot:
    image: localhost:5005/iot20/g3a/pi
    volumes: 
      - /dev/mem:/dev/mem
      - /dev/gpiomem:/dev/gpiomem
    deploy:
      placement:
        constraints:
          # Place it on the rpi
          - node.labels.rpi == true
    labels:
      - "volume.device=/dev/mem"
      - "volume.device=/dev/gpiomem"

**Comment by mnestor** [comment]
I only have this working for 1 device per service right now.

**Comment by gevzak** [comment]
Thanks for your prompt reply.
I tried reducing it to one device, to /dev/mem to be exact, but it still didn't work sadly. 
Does the config look right?

**Comment by mnestor** [comment]
yes, i did just push up a change. I found an issue that you needed do mount /dev in order to resolve symbolic links to devices.

**Comment by gevzak** [comment]
Thank you very much for your commit.
Sadly, i still cannot get it to work. Your container gets the event, starts to set permissions but my docker fails to access to /dev.
I suspect that the reason might be me not deploying it correctly.
Can you please explain a bit in details how should i deploy and start them?
Once again, thank you for your time.

**Comment by mnestor** [comment]
I see that you have it loaded right since it is trying to grant permission for your container. Did you load the stack yaml as is or did you modify anything in it? The stack needs to start a local container with full permissions in order to set permissions for your container that will access /dev/mem? It could also be that the level it needs to open that device is beyond what this is able to do. I really don't understand it that much, it's just cut up snippets of other peoples comments or specific settings they used for a specific container.
